train.py

```diff
@@ -0,0 +1,164 @@
+import uuid
+import itertools
+from typing import Callable
+
+import hydra
+import torch.optim
+import torch.nn as nn
+from tqdm import tqdm
+from omegaconf import DictConfig
+from torch.utils.data import DataLoader
+
+import wandb
+from model import PitchSeqNN
+from data.dataset import BagOfPitches
+
+
+@hydra.main(version_base=None, config_path="conf", config_name="config")
+def main(cfg: DictConfig):
+    # loading data
+    dataset = BagOfPitches()
+    v_dataset = BagOfPitches(split="validation")
+    train_dataloader = DataLoader(dataset, batch_size=cfg.hyperparameters.batch_size, shuffle=True)
+    v_dataloader = DataLoader(v_dataset, batch_size=cfg.hyperparameters.batch_size)
+
+    # get composers to classify against each other
+    count = dataset.df.groupby(["composer"]).size()
+    composers_with_most_data = count[count.values > 10]
+    composers_to_check = composers_with_most_data.index.tolist()
+    composers = [pair for pair in itertools.combinations(composers_to_check, r=2)]
+    print(f"pairs to check: {len(composers)}")
+
+    # container for trained models
+    models = []
+
+    # train model for each pair
+    for pair in composers:
+        print(f"{pair[0]} vs {pair[1]}")
+        model = run_experiment(cfg=cfg, train_data=train_dataloader, val_data=v_dataloader, classnames=pair)
+        models.append((model, pair))
+        path = f"models/{pair[0].replace(' ', '_').lower()}-{pair[1].replace(' ', '_').lower()}.pth"
+        if cfg.model.save_model == 'y':
+            torch.save(model.state_dict(), path)
+
+
+def run_experiment(cfg: DictConfig, train_data: DataLoader, val_data: DataLoader, classnames: list):
+    """
+    Run an experiment using the provided configuration and data.
+
+    Parameters:
+        cfg (DictConfig): Configuration containing hyperparameters and model specifications.
+        train_data (DataLoader): DataLoader for the training dataset.
+        val_data (DataLoader): DataLoader for the validation dataset.
+        classnames (list[str]): A list containing the class names corresponding to their indices.
+
+    Returns:
+        PitchSeqNN: The trained PitchSeqNN model.
+
+    This function initializes an experiment on WandB, initializes the model and optimizer, and performs
+    the training and validation loops. It logs the training and validation statistics to WandB.
+    """
+    run_id = str(uuid.uuid1())[:8]
+    run = wandb.init(
+        project="MIDI-18-bag-of-pitches",
+        name=f"{classnames[0].replace(' ', '_')}_vs_{classnames[1].replace(' ', '_')}-{run_id}",
+        config={
+            "learning_rate": cfg.hyperparameters.lr,
+            "n_epochs": cfg.hyperparameters.num_epochs,
+            "architecture": "NN",
+            "batch_size": cfg.hyperparameters.batch_size,
+        },
```

Comment:
```suggestion
        config=OmegaConf.to_container(cfg, resolve=True),
```

We can store the full hydra config in wandb